jscomp/runtime/caml_obj.ml

```diff
@@ -148,11 +153,13 @@ let caml_update_dummy (x : Caml_obj_extern.t) (y : Caml_obj_extern.t) : unit =
   done ; 
   let y_tag = Caml_obj_extern.tag y in 
   if y_tag <> 0 then
-    Caml_obj_extern.set_tag x y_tag
+    Caml_obj_extern.set_tag x y_tag *)
 (* Caml_obj_extern.set_length x   (Caml_obj_extern.length y) *)
 (* [set_length] seems redundant here given that it is initialized as an array 
 *)
 
+external caml_update_dummy : Caml_obj_extern.t -> Caml_obj_extern.t -> unit = "Object.assign" [@@bs.val]
+
```

Comment:
It seems we need tell difference between objects and arrays here, I will fix it

Reply:
In latest master branch, caml_update_dummy should handle this properly for both array and record

Reply:
done